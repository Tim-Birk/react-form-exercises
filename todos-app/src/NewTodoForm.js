import React, { useState } from 'react';

const NewTodoForm = ({ addTodo }) => {
  const intialState = {
    task: '',
  };
  const [formData, setFormData] = useState(intialState);

  const handleChange = (evt) => {
    const { name, value } = evt.target;
    setFormData((fData) => ({
      ...fData,
      [name]: value,
    }));
  };

  const handleSubmit = (evt) => {
    evt.preventDefault();
    const { task } = formData;
    addTodo(task);
    setFormData(intialState);
  };

  return (
    <form onSubmit={handleSubmit}>
      <div>
        <label htmlFor='task'>Task:</label>
        <input
          id='task'
          name='task'
          value={formData.task}
          onChange={handleChange}
          placeholder='Enter a task'
        />
      </div>
      <button>Add New</button>
    </form>
  );
};

export default NewTodoForm;
