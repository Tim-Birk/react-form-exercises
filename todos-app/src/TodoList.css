.TodoList {
  padding-left: 0;
  margin-top: 10px;
}
