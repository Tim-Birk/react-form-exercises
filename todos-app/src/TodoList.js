import React, { useState } from 'react';
import Todo from './Todo';
import NewTodoForm from './NewTodoForm';
import { v4 as uuid } from 'uuid';
import './TodoList.css';

const TodoList = () => {
  const [Todos, setTodos] = useState([
    { id: uuid(), task: 'Walk the dog' },
    { id: uuid(), task: 'Make breakfast' },
    { id: uuid(), task: 'Study coding' },
    { id: uuid(), task: 'Exercise' },
  ]);

  const addTodo = (task) => {
    const newTodo = { id: uuid(), task };
    setTodos((oldTodos) => [...oldTodos, newTodo]);
  };

  const removeTodo = (id) => {
    setTodos(Todos.filter((Todo) => Todo.id !== id));
  };

  return (
    <>
      <h1>Todo List</h1>
      <NewTodoForm addTodo={addTodo} />
      <ul className='TodoList'>
        {Todos.map((t) => (
          <Todo key={t.id} id={t.id} task={t.task} removeTodo={removeTodo} />
        ))}
      </ul>
    </>
  );
};

export default TodoList;
