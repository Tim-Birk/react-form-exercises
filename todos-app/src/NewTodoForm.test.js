import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';
import TodoList from './TodoList';

it('renders without crashing', function () {
  render(<NewTodoForm />);
});

it('matches snapshot', function () {
  const { asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});

it('can add a new Todo', function () {
  const { getByLabelText, queryByText } = render(<TodoList />);

  // no Todo yet
  expect(queryByText('- This is a test')).not.toBeInTheDocument();

  const taskInput = getByLabelText('Task:');
  const submitBtn = queryByText('Add New');

  // fill out the form
  fireEvent.change(taskInput, { target: { value: 'This is a test' } });
  fireEvent.click(submitBtn);

  // Todo exists!
  expect(queryByText('- This is a test')).toBeInTheDocument();
});
