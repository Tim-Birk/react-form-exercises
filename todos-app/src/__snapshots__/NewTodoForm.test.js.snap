// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <div>
      <label
        for="task"
      >
        Task:
      </label>
      <input
        id="task"
        name="task"
        placeholder="Enter a task"
        value=""
      />
    </div>
    <button>
      Add New
    </button>
  </form>
</DocumentFragment>
`;
